File: java/src/main/java/IMovieService.java
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;

public interface IMovieService {

    JSONObject getAllMovies() throws IOException, JSONException;

    JSONObject getCurrentRentals(String username) throws IOException, JSONException;

    JSONObject rentMovie(String username, Integer movieId) throws IOException, JSONException;

    JSONObject addMovie(String username, Movie movie) throws IOException, JSONException;
}

File: c_sharp/MockingExample/App/IMovieService.cs
using System;
using System.Collections.Generic;
using TDD_Mocking_Workshop;

namespace MovieRentalApp
{
    public interface IMovieService
    {
        String GetAllMovies();
        String GetCurrentRentals(String username);
        String RentMovie(String username, int movieId);
        String AddMovie(String username, Movie movie);

    }
}
